lib/src/browser_client.dart

```diff
@@ -50,34 +67,36 @@ class BrowserClient extends BaseClient {
       ..responseType = 'arraybuffer'
       ..withCredentials = withCredentials;
     request.headers.forEach(xhr.setRequestHeader);
-
-    var completer = Completer<StreamedResponse>();
+    if (timeout != null) {
+      onTimeout = () {
+        xhr.abort();
```

Comment:
Should we be removing the xhr from _xhrs here? It seems to me that currently that would only happen once the request completes with either an error or success. Presumably calling abort does make it complete with an error, but then i would only expect to see the call to `abort` here, otherwise we are likely trying to complete the completer twice?

Reply:
Maybe we should remove the xhr whenever the completer completes actually instread of doing it manually in all these places?

Reply:
Seems like we remove it on `completer.future.whenComplete` above, so it should be fine.